_drafts/oryx/packages.md

```diff
@@ -0,0 +1,99 @@
+
+## Layers
+
+While packages are all distributed as a flat list, there is an architectural hierarchy. The hierarchy is important as it will protect from cyclic dependencies. Packages inside a layer can depend on sibling packages inside the layer without any issues. Packages can never depend on a layer above.
+
+While the package layering might not be relevant during your development, it might help you to better understand the package dependencies. The diagram below shows you the four package layers, the top layer in the diagram is the [boilerplate application](./boilerplate.md) that uses a preset to setup the a standard application.
+
+<!-- Add link to presets.md (see https://spryker.atlassian.net/browse/HRZ-2153) -->
+
+```mermaid
+%%{init: {'theme':'base', 'themeVariables': { 'primaryColor': '#fff','primaryBorderColor': '#aaa'}}}%%
+
+flowchart TD
+classDef app fill:#fff,stroke:#fff;
+
+app("Application"):::app
+template(Template packages):::template
+domain(Domain packages)
+platform(Platform packages)
+base(Base packages)
+
+app-..->|"use presets\n(e.g. b2c)"|template
+app-->domain
+
+template --> domain
+domain --> platform
+platform --> base
+domain --> base
+```
+
+### Template packages
+
+The template layer contains packages that can be used as quick starters for demos and projects. Templated packages do follow semantic versioning and will deliver on the promise of upgradability, but are fairly opinionated as they mainly serve to get up to speed fast with a standard frontend application.
+
+| Packages                                                       | Location                    |
+| -------------------------------------------------------------- | --------------------------- |
+|                                                                |                             |
+| [Application](https://www.npmjs.com/package/@spryker-oryx/)    | `@spryker-oryx/application` |
+| [Presets](https://www.npmjs.com/package/@spryker-oryx/presets) | `@spryker-oryx/presets`     |
+| [Labs ](https://www.npmjs.com/package/@spryker-oryx/labs)      | `@spryker-oryx/labs`        |
+| [Themes ](https://www.npmjs.com/package/@spryker-oryx/themes)  | `@spryker-oryx/themes`      |
+
+> **Note:** the labs package is an exception here. The labs package consists of experimental or demo functionality and is not recommended to be used in production.
+
+### Domain packages
+
+Domain packages provide components, services and adapters for a certain domain. Organising packages in domains helps the developer experience, as its' fairly easy to understand where to find a certain component or service. An example of a domain package is the `product` package, which contains all product related components as well as the product services and adapters that integrates with Spryker APIs.
```

Comment:
> Domain packages provide components, services and adapters for a certain domain.

Maybe we can generalise it somehow, as long as domain package can contain something else then components/services/adapters. So we do we want to limit it? Or we can at least add `etc.`